File: app/src/main/java/kz/mycrm/android/ui/info/ServiceAdapter.java
package kz.mycrm.android.ui.info;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import kz.mycrm.android.R;

/**
 * Created by asset on 12/7/17.
 */

public class ServiceAdapter extends RecyclerView.Adapter<ServiceAdapter.ViewHolder>{

    List<ServiceObj> serviceList;
    Context context;

    public ServiceAdapter(List<ServiceObj> serviceList, Context context) {
        this.serviceList = serviceList;
        this.context = context;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(context).inflate(R.layout.item_service, parent, false);

        ViewHolder vh = new ViewHolder(view);

        return vh;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.price.setText(serviceList.get(position).price);
        holder.title.setText(serviceList.get(position).title);
    }

    @Override
    public int getItemCount() {
        return serviceList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{

        @BindView(R.id.service_title)
        TextView title;
        @BindView(R.id.service_price)
        TextView price;

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(itemView);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Toast.makeText(context, "service number " + itemView.getId(), Toast.LENGTH_SHORT).show();
                }
            });
        }
    }
}

File: app/src/main/java/kz/mycrm/android/ui/main/info/ServiceAdapter.kt
package kz.mycrm.android.ui.main.info

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import kz.mycrm.android.R
import kz.mycrm.android.db.entity.Service

/**
 * Created by asset on 12/7/17.
 */

class ServiceAdapter(private var serviceList: List<Service>, internal var context: Context) : RecyclerView.Adapter<ServiceAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_service, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val service = serviceList[position]
        holder.price.text = service .price
        holder.title.text = service .serviceName
    }

    override fun getItemCount(): Int {
        return serviceList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        internal var title: TextView = itemView.findViewById<View>(R.id.service_title) as TextView
        internal var price: TextView = itemView.findViewById<View>(R.id.service_price) as TextView

        init {
//            itemView.setOnClickListener { Toast.makeText(context, "service number " + itemView.id, Toast.LENGTH_SHORT).show() }
        }
    }
}
